src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverter.cs

```diff
@@ -0,0 +1,252 @@
+                        {
+                            case double d:
+                                s = d.ToString("G");
+                                break;
+
+                            case decimal dm:
+                                s = dm.ToString("G");
+                                break;
+
+                            case float f:
+                                s = f.ToString("G");
+                                break;
+                        }
+                        @long = (long)ldv;
+                        @double = ldv.ToDouble(CultureInfo.InvariantCulture);
+                        scope.Write(field.Id, s, @long, @double, ref entryWriter);
+                        return;
+                    }
+                    
+                case ValueType.Numeric:
+                    var lazyNumber = value as LazyNumberValue;
+                    if (lazyNumber == null)
+                    {
+                        scope.Write(field.Id, lazyNumber.Inner.AsSpan(), (long)value, Convert.ToDouble(value), ref entryWriter);
+                        return;
+                    }
+                    @long = (long)lazyNumber;
+                    @double = lazyNumber.ToDouble(CultureInfo.InvariantCulture);
+
+                    scope.Write(field.Id, lazyNumber.Inner.AsSpan(), @long, @double, ref entryWriter);
+                    return;
+
+                case ValueType.String:
+                    scope.Write(field.Id, (string)value, ref entryWriter);
+                    return;
+
+                case ValueType.LazyCompressedString:
+                case ValueType.LazyString:
+                    LazyStringValue lazyStringValue;
+                    if (valueType == ValueType.LazyCompressedString)
+                        lazyStringValue = ((LazyCompressedStringValue)value).ToLazyStringValue();
+                    else
+                        lazyStringValue = (LazyStringValue)value;
+                    scope.Write(field.Id, lazyStringValue.AsSpan(), ref entryWriter);
+                    return;
+
+                    case ValueType.Enum:
+                    scope.Write(field.Id, value.ToString(), ref entryWriter);
+                    return;
+
+                case ValueType.Boolean:
+                    scope.Write(field.Id, (bool)value ? _trueLiteral : _falseLiteral, ref entryWriter);
+                    return;
+
+                case ValueType.DateTime:
+                    var dateTime = (DateTime)value;
+                    var dateAsBytes = dateTime.GetDefaultRavenFormat();
+                    scope.Write(field.Id, dateAsBytes, ref entryWriter);
+                    return;
+
+                case ValueType.DateTimeOffset:
+                    var dateTimeOffset = (DateTimeOffset)value;
+                    var dateTimeOffsetBytes = dateTimeOffset.UtcDateTime.GetDefaultRavenFormat(isUtc: true);
+                    scope.Write(field.Id, dateTimeOffsetBytes, ref entryWriter);
+                    return;
```

Comment:
In Lucene, we index date times twice - once as a string, and once numeric `Ticks`.
I don't think we need to do it here at this stage, but important difference to note